src/test/java/org/jitsi/meet/test/web/WebParticipantFactory.java

```diff
@@ -322,4 +350,24 @@ private URL getRemoteDriverAddress()
 
         return null;
     }
+
+    /**
+     * Downloads extension from chrome webstore.
+     * @param id - the id of the extension.
+     * @returns <tt>File</tt> associated with the downloaded extension.
+     */
+    private static File downloadExtension(String id)
+        throws IOException
+    {
+        URL website = new URL(
+            EXT_DOWNLOAD_URL_PREFIX + id + EXT_DOWNLOAD_URL_SUFIX);
+        ReadableByteChannel rbc = Channels.newChannel(website.openStream());
+        File extension = File.createTempFile("jidesha", ".crx");
+        extension.deleteOnExit();
+
+        FileOutputStream fos = new FileOutputStream(extension);
```

Comment:
use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Sorry, my java is a little bit rusty as I said. So could you please explain with more details? The part that puzzles me is I want this method to trow the exception.

Reply:
yes, but the purpose of using that is to close the resource, not to catch the IOException

Reply:
OK. I'll read the docs more carefully!